spring-integration-kafka/src/main/java/org/springframework/integration/kafka/inbound/KafkaInboundGateway.java

```diff
@@ -76,7 +74,7 @@
  *
  */
 public class KafkaInboundGateway<K, V, R> extends MessagingGatewaySupport
-		implements KafkaInboundEndpoint, Pausable, OrderlyShutdownCapable {
```

Comment:
I don't see a reason in this change. It is definitely a contract for endpoints, but not specific listener.
Would you mind explaining why did you go this way?

Reply:
Yes, you are right. But, if we do that, `ATTRIBUTES_HOLDER` become publicly accessible via `KafkaMessageDrivenChannelAdapter#getAttributesHolder` method, and that is not good in terms of Encapsulation. So I tried to hide it from users.
Another option could be to move the methods within KafkaInboundEndpoint into MessagingMessageListenerAdapter or somewhere else.

Anyway, it might not be a big deal at the moment.